internal/clients/team4/iifo.go

```diff
@@ -0,0 +1,150 @@
+
+	// Implement the variance cap chosen
+	if averageVariance > varianceLimit {
+		averageVariance = varianceLimit
+	}
+
+	// Return the confidence of the prediction
+	return math.Round(varianceLimit - averageVariance)
+}
+
+// ReceiveDisasterPredictions provides each client with the prediction info, in addition to the source island,
+// that they have been granted access to see
+// COMPULSORY, you need to implement this method
+func (c *client) ReceiveDisasterPredictions(receivedPredictions shared.ReceivedDisasterPredictionsDict) {
+	// If we assume that we trust each island equally (including ourselves), then take the final prediction
+	// of disaster as being the weighted mean of predictions according to confidence
+	numberOfPredictions := float64(len(receivedPredictions) + 1)
+	predictionInfo := c.obs.iifoObs.ourDisasterPrediction.PredictionMade
+	selfConfidence := predictionInfo.Confidence
+
+	// Initialise running totals using our own island's predictions
+	totalCoordinateX := selfConfidence * predictionInfo.CoordinateX
+	totalCoordinateY := selfConfidence * predictionInfo.CoordinateY
+	totalMagnitude := selfConfidence * predictionInfo.Magnitude
+	totalTimeLeft := uint(math.Round(selfConfidence)) * predictionInfo.TimeLeft
+	totalConfidence := selfConfidence
+
+	// Add other island's predictions using their confidence values
+	for _, prediction := range receivedPredictions {
+		totalCoordinateX += prediction.PredictionMade.Confidence * prediction.PredictionMade.CoordinateX
+		totalCoordinateY += prediction.PredictionMade.Confidence * prediction.PredictionMade.CoordinateY
+		totalMagnitude += prediction.PredictionMade.Confidence * prediction.PredictionMade.Magnitude
+		totalTimeLeft += uint(math.Round(prediction.PredictionMade.Confidence)) * prediction.PredictionMade.TimeLeft
+		totalConfidence += prediction.PredictionMade.Confidence
+	}
+
+	// Finally get the final prediction generated by considering predictions from all islands that we have available
+	// This result is currently unused but would be used in decision making in full implementation
+	finalPrediction := shared.DisasterPrediction{
+		CoordinateX: totalCoordinateX / totalConfidence,
+		CoordinateY: totalCoordinateY / totalConfidence,
+		Magnitude:   totalMagnitude / totalConfidence,
+		TimeLeft:    uint((float64(totalTimeLeft) / totalConfidence) + 0.5),
+		Confidence:  totalConfidence / numberOfPredictions,
+	}
+	c.obs.iifoObs.finalDisasterPrediction = finalPrediction
+	c.Logf("Final Prediction: [%v]", finalPrediction)
+}
+
+// MakeForageInfo allows clients to share their most recent foraging DecisionMade, ResourceObtained from it to
+// other clients.
+// OPTIONAL. If this is not implemented then all values are nil.
+func (c *client) MakeForageInfo() shared.ForageShareInfo {
+	contribution := shared.ForageDecision{Type: shared.DeerForageType, Contribution: 0}
+	return shared.ForageShareInfo{DecisionMade: contribution, ResourceObtained: 0, ShareTo: []shared.ClientID{}}
+}
```

Comment:
so we're always returning that we went deer hunting, even though there is 50% chance that we went fishing?
shouldn't we be truthful and report what we did (to all the islands, or the ones who we trust), so that other islands who actually take it into consideration can make more educated decisions?

Reply:
Sorry, this is the baseclient implementation, Toby will overwrite this